openlabcmd/openlabcmd/plugins/nodepool/orphanresource.py

```diff
@@ -0,0 +1,120 @@
+import copy
+import subprocess
+
+from openlabcmd.plugins.base import Plugin
+
+SERVER_WHITE_LIST = [
+    '%s-openlab-zuul', '%s-openlab-nodepool', '%s-openlab-zookeeper']
+
+
+class OrphanResourcePlugin(Plugin):
+
+    # This Plugin is only for checking, as it's a dangerous action for each.
+    ptype = 'nodepool'
+    name = 'OrphanResource'
+
+    def check(self):
+        self.failed = False
+        if self.reasons:
+            self.reasons.pop()
+        server_white_list = [elemnt % self.cloud
+                             for elemnt in copy.deepcopy(SERVER_WHITE_LIST)]
+
+        # For floatingip, we will check whether a fip didn't associated with a
+        # neutron port.
+        get_fip_cmd = ("openstack --os-cloud %s floating ip list "
+                       "-c 'ID' -c 'Floating IP Address' -c 'Port' | "
+                       "egrep '[a-z0-9]{8}-[a-z0-9]{4}-[a-z0-9]{4}-"
+                       "[a-z0-9]{4}-[a-z0-9]{12}' | awk '{print $2, $4, $6}' "
+                       "| grep None | awk '{print $1}'")
+        # For volume, we just check the available status volumes, as we cannot
+        # know whether a in-use volume is used for any purpose.
+        get_volume_cmd = (
+            "openstack --os-cloud %s volume list -c 'ID' -c 'Status' | "
+            "egrep '[a-z0-9]{8}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{4}-"
```

Comment:
`openstack volume list --status available -f value -c ID`

Reply:
and I want to remember, we may shouldn't treat all the volumes in `available` status as orphan volume. for some of the `cloud-provider-openstack` jobs, it will try to create volumes in cloud as PVs, when running jobs, the jobs may create volumes, and before attaching, the volume maybe in `available` status

Reply:
So, if fip is in the same case, the fip not attach a real port during test. How about add another filter? Check the created_at.

Reply:
yes, so maybe what we can do is only to audit the orphan resource,  and then let operators to decide to delete or not..